src/timeline/TimelineModel.cpp

```diff
@@ -741,6 +774,46 @@ TimelineModel::internalAddEvents(
 
                         if (encInfo)
                                 emit newEncryptedImage(encInfo.value());
+
+                        if (emitCallEvents) {
+                                // event room_id is not set, apparently due to spec bug
+                                if (auto callInvite = std::get_if<
+                                      mtx::events::RoomEvent<mtx::events::msg::CallInvite>>(&e_)) {
+                                        callInvite->room_id = room_id_.toStdString();
+                                        emit newCallEvent(e_);
+                                } else if (std::holds_alternative<mtx::events::RoomEvent<
+                                             mtx::events::msg::CallCandidates>>(e_) ||
+                                           std::holds_alternative<
+                                             mtx::events::RoomEvent<mtx::events::msg::CallAnswer>>( e_) ||
+                                           std::holds_alternative<
+                                             mtx::events::RoomEvent<mtx::events::msg::CallHangUp>>( e_)) {
+                                        emit newCallEvent(e_);
+                                }
+                        }
+                }
+
+                if (std::holds_alternative<
+                      mtx::events::RoomEvent<mtx::events::msg::CallCandidates>>(e)) {
+                        // don't display CallCandidate events to user
+                        events.insert(id, e);
+                        if (emitCallEvents)
+                                emit newCallEvent(e);
+                        continue;
+                }
+
+                if (emitCallEvents) {
+                        // event room_id is not set, apparently due to spec bug
```

Comment:
That's actually because /sync returns events without room_id so the id is only set on some events.